cranelift/frontend/src/frontend.rs

```diff
@@ -596,10 +596,11 @@ impl<'a> FunctionBuilder<'a> {
         }
     }
 
-    /// Declare that translation of the current function is complete. This
-    /// resets the state of the `FunctionBuilder` in preparation to be used
-    /// for another function.
-    pub fn finalize(&mut self) {
+    /// Declare that translation of the current function is complete.
+    ///
+    /// This resets the state of the `FunctionBuilderContext` in preparation to
+    /// be used for another function.
+    pub fn finalize(mut self) {
```

Comment:
cg_clif uses `fx.builder.finalize()` in a function taking `fx` by mutable reference. As such moving out is not allowed. It should be possible to change this though.